### ISSUE TITLE
GSUB corruption

### ISSUE THREAD

**Issue by bobh0303** [silnrsi/font-ttf#13]
This program copies a font and, in the process, reads and rebuilds GSUB and GPOS tables:
```
#!/usr/bin/perl
#
use Font::TTF::Font;
$font = Font::TTF::Font->open($ARGV[0]);
$font->{'GSUB'}->read;
$font->{'GPOS'}->read;
$font->out($ARGV[1]);
```
For some fonts, however, the resulting GSUB and/or GPOS appear to be corrupted. We first noticed this because Glyphs wouldn't open certain TTF files coming out of TypeTuner. 

Comparing the GSUBs of the original ([Lateef-Regular.ttf 2.000](https://software.sil.org/downloads/r/lateef/Lateef-2.000.zip), upper window) and the copy (lower window) using Font Validator (the OpenSource one, v2.1.6) yields:

![image](https://user-images.githubusercontent.com/6842108/186300105-3abed6ee-b57a-42b0-9c29-39a5cd91b522.png)

Using ttx to dump the GSUB tables of these same two fonts, then comparing... the first difference is at Lookup 20, Subtable 1:

![image](https://user-images.githubusercontent.com/6842108/186301318-df4c097e-d5ec-46b0-8b2f-a001b8e8026e.png)
(which, yeah, is the same information as from FontValidator).

I have not had a chance to dig much further, but these look serious enough that I expect the fonts will not function as we intended.

**Comment by bobh0303** [comment]
I have verified that even if these changes do not prevent Windows from using the font, the font is functionally incorrect, i.e., not working as intended.

**Comment by bobh0303** [comment]
After further research, there are two kinds of corruption afaict:

- context lookups have `SourceIndex` or `LookupListIndex` modified to out of range values.
- The script records get corrupted, sometimes having huge numbers of clearly bogus Language Systems records.

**Comment by bobh0303** [comment]
Fixed with 14103bc977c52f8999e2c8ab477c9ce85824ba8e and a884cc8dc0274ddc60d64a6e81e4b3f26e72b5fd respectively.
